README.md

```diff
@@ -1 +1,122 @@
 # ChunkedBase.jl
+
+The package handles the ingestion of data chunks and the distribution & synchronization of work that happens on these chunks in parallel. It came into existence while refactoring the [`ChunkedCSV.jl`](https://github.com/RelationalAI/ChunkedCSV.jl) and [`ChunkedJSONL.jl`](https://github.com/RelationalAI/ChunkedJSONL.jl) packages and was designed to be extended by packages like these. It is a package used to write parser packages.
```

Comment:
I think the README would benefit from starting with _why this package exists_  (what it's aiming to achieve / what the design constraints are), rather than _what it does_ 

(this could be a "goals" section, before a "design / how it works" section, before a "example / how to use it" section. Non-goals might be interesting too  e.g. [LibPQ.jl readme](https://github.com/iamed2/LibPQ.jl#goals) has these. 

e.g. rather than 
> The package handles the ingestion of data chunks

something like
> The package aims to ingest "chunkable" data with maximum throughput. By "chunkable" data we mean any data that [blah]. For example, CSV and JSONLines are both chunkable because [something about newlines], and the ChunkedCsv.jl and ChunkedJsonl.jl use this package to [achieve cool perf thing].
> Additionally, ChunkedBase.jl aims to...
> - never abort on invalid record (? is this a thing that ChunkedBase does? or ChunkedCSV? or our raicode `consume!` method? or?)
> - etc.

but, y'know, better written than i can manage at this time of night

Reply:
I don't understand why is it better to talk about goals rather than the things the package does. What is missing from the current description?

Reply:
ah, because otherwise it's hard to understand why things are the way they are

or if it would be appropriate to make certain changes?

e.g. (silly example) imagine a future maintainer wants to drop `parse_parallel` because it seems complicated and `parse_serial` gives the same results, why wouldn't that be a welcome change?

or, maybe less silly, what if they wanted to throw errors more often?

or, a potential user what to try to use this to build a JSON parser, how would they know if ChunkedBase.jl is a good tool for that?

right now, i feel like the README doesn't help get me thinking about things the same way that you and Jacob etc were thinking at the time you developed this package

put another, it feels more like memorising how the package works, not understanding/internalising it

does that make sense?

Reply:
Thanks Nick, I think the latest changes improved this aspect a bit